docs/spec.rst

```diff
@@ -0,0 +1,501 @@
+JSpecify 0.1 spec outline
+=========================
+
+About this document
+-------------------
+
+This document summarizes our current thinking for proposing a set of
+nullness annotations. **Nothing here is “set in stone” yet.**
+
+It will be very helpful to read or refer to the `core
+concepts/glossary <https://docs.google.com/document/d/1KQrBxwaVIPIac_6SCf--w-vZBeHkTvtaqPSU_icIccc/edit>`__
+document in order to understand this material correctly. There are
+numerous other documents in our `shared
+folder <https://drive.google.com/drive/folders/1vZl1odNCBncVaN7EwlwfqI05T_CHIqN->`__.
+
+Overview
+--------
+
+We propose a set of type annotations to specify the intended nullness of
+individual type usages. At a high level, our proposed semantics is
+similar to existing tools’ treatment of nullness type annotations. We
+treat nullness annotations as purely compile-time type refinements
+without effect on runtime semantics. Besides the typical “nullable” and
+“non-null” refinements, we distinguish values of “unspecified nullness”,
+which typically arise from unannotated code, and allow for (with
+optional warnings if desired) unsound subtyping between “unspecified
```

Comment:
We should probably not call it subtyping.  how about "unsound coercions" or "unsound conversions" (as in the proposal doc)?

Reply:
I worried about using "conversions" because the JLS idea of "conversions" is more specific that what we're aiming for. For example, we want for an unannotated `class Foo<T>` to be instantiable as `Foo<@Nullable Object>`. As I understand it from [JLS 4.5.1](https://docs.oracle.com/javase/specs/jls/se14/html/jls-4.html#jls-4.5.1), the rule that's applied there is containment (which applies subtyping), so it doesn't technically involve "conversion."

"Coercion" sounds workable, especially since I don't see that word anywhere in the JLS. Regrettably, I skimmed over that part of your message original, so I went to some contortions instead. Let me know what you think, and I can go with coercion if you prefer that.

Reply:
SGTM.  We can separately try to work on this some more in the underlying doc